source/reference/array.rst

```diff
@@ -236,7 +236,136 @@ Unevaluated Items
+more in certain situations. ("Closed" to two arguments in some
+places, "open" to more arguments when you need it to be.)
+
+.. schema_example::
+
+    {
+      "$id": "https://example.com/my-tuple",
+
+      "type": "array",
+      "items": [
+        true,
+        { "type": "boolean" }
+      ],
+
+      "$defs": {
+        "closed": {
+          "$anchor": "closed",
+          "$ref": "#"
+        }
+      }
+    }
+
+``items`` evaluates only the items in the initial array. Any items
+you add when you extend the tuple later *are not evaluated*. When
+you extend the tuple, the schema passes validation only if you include
+``unevaluatedItems`` like this:
+
+.. schema_example::
+
+    {
+      "$id": "https://example.com/my-extended-tuple",
+
+      "$ref": "https://example.com/my-tuple",
+      "prefixItems": [
+        true,
+        true,
+        { "type": "boolean" }
+      ],
+      "unevaluatedItems": false
+    }
+
+With this, you can use ``$ref`` to reference the first two
+``prefixItems`` and keep the schema "closed" to two arguments when
+you need it, "open" to more arguments when you need it. A reference to
+``/my-tuple#closed`` would disallow more than two items. And because
+``unevaluatedItems`` only sees inside its own subschema, if you
+want to add an item, add it inside that subschema.
+
+This means you can also put ``unevaluatedItems`` in a nested tuple.
+
+.. schema_example::
+
+    {
+      "prefixItems": [
+        { "type": "string" }
+      ],
+      "allOf": [
+        {
+          "prefixItems": [
+            true,
+            { "type": "number" }
+          ]
+        }
+      ],
+      "unevaluatedItems": false
+    }
+    --
+    ["foo", 42]
+    // All the array items are evaluated. The schema passes validation.
+    --X
+    ["foo", 42, null]
+    // The third value is unevaluated, so ``unevaluatedItems`` is true,
+    // not false. The schema fails validation.
+
+Note these two restraints, though. First, ``items`` doesn't "see
+inside" any instances of ``allOf``, ``anyOf``, or ``oneOf`` in the
+same subschema. In this next example, ``items`` ignores ``allOf`` and
+thus fails to validate.
+
+.. schema_example::
+
+    {
+      "allOf": [{ "prefixItems": [{ "enum": [1, "a"] }]  }],
+      "items": { "const": 2 }
+    }
+    --X
+    [1, "a", 2]
+
+Second, ``unevaluatedItems`` can't "see inside" cousins (vertically
+adjacent properties inside a separate pair of {curly braces} with the
+same "parent"— ``anyOf``, ``if``, ``not``, or similar). For instance,
+in the example below, the ``unevaluatedItems`` doesn't "see inside"
+the ``prefixItems`` cousin before it. All instances fail validation
+because ``"prefixItems": [ true ]`` matches only length 1 arrays, and
+``{ "unevaluatedItems": false }`` matches only empty arrays.
+
+.. schema_example::
+
+    {
+      "allOf": [
+        { "prefixItems": [true] },
+        { "unevaluatedItems": false }
+      ]
+    }
+    --X
+    [1]
```

Comment:
I'm not sure this needs to be included. It doesn't describe how the keyword works as much as how it doesn't work. If this was something that we frequently see people misunderstanding, it think it would be worth including, but I haven't seen that to be the case. I think this example can be left out.